packages/squiggle-lang/src/fr/list.ts

```diff
@@ -365,20 +357,12 @@ export const library = [
     examples: [`List.slice([1,2,5,10],1,3)`],
     definitions: [
       makeDefinition(
-        [frArray(frGeneric("A")), frNumber],
-        frArray(frGeneric("A")),
-        ([array, start]) => {
-          _assertInteger(start);
-          return array.slice(start);
-        }
-      ),
-      makeDefinition(
-        [frArray(frGeneric("A")), frNumber, frNumber],
+        [frArray(frGeneric("A")), frNumber, frOptional(frNumber)],
         frArray(frGeneric("A")),
         ([array, start, end]) => {
           _assertInteger(start);
-          _assertInteger(end);
-          return array.slice(start, end);
+          end && _assertInteger(end);
+          return end ? array.slice(start, end) : array.slice(start);
```

Comment:
I can imagine that `end` could be `0` here, that might cause an incorrect behavior (not sure).

So I'd suggest `end === null` check; but also see my comment on `fr/calculator.ts` about null vs undefined.